plugins/pipelines/startAll.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.pipeline.base, 'id');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/pipelines/{id}/startall',
+    config: {
+        description: 'Start all child pipelines given a specific pipeline',
+        notes: 'Start all child pipelines given a specific pipeline',
+        tags: ['api', 'pipelines'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
+            const eventFactory = request.server.app.eventFactory;
+            const userFactory = request.server.app.userFactory;
+            const id = request.params.id;
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+            const scm = pipelineFactory.scm;
+
+            return Promise.all([
+                pipelineFactory.get(id),
+                userFactory.get({ username, scmContext })
+            ])
+                .then(([pipeline, user]) => {
```

Comment:
do we need to check if user exists?
```
if (!user) {
     throw boom.notFound(`User ${username} does not exist`);
}
```

Reply:
We have that in a lot of our endpoints, but i forgot why we need it.. We're getting `username` and `scmContext` from user's JWT, how is it possible that the user doesn't exists?

Reply:
I see what you're saying. Should be okay to not have this check then.